mock_test.go

```diff
@@ -343,15 +343,19 @@ func TestConsumerRequeueNoBackoff(t *testing.T) {
 		"RDY 0",
 		fmt.Sprintf("REQ %s 0", msgIDRequeueNoBackoff),
 		"RDY 1",
-		"RDY 1",
+		"RDY 1",   // Duplicate rdy update will be ignored.
```

Comment:
seems like, if you just comment out this item completely, none of the additional changes below are needed

Reply:
@ploxiln Sorry for not understanding your opinion.
> if you just comment out this item completely, none of the additional changes below are needed

Yes. But i think we should test the deduplicate rdy update logic.

Reply:
This line is not for the input, this is line is for the expected output. Commenting out this line does not change what the "rdy update logic" goes through in this test. Commenting out this line is equivalent to, but simpler, than the code added just below to skip over this line.

Reply:
@ploxiln Thanks for your tips. The code has been updated. PTAL.

Reply:
@ploxiln PTAL.

Reply:
/cc @jehiah